presto-main/src/main/java/com/facebook/presto/sql/gen/BytecodeUtils.java

```diff
@@ -181,45 +180,85 @@ public static BytecodeNode generateInvocation(Scope scope, String name, ScalarFu
         // Index of parameter (without @IsNull) in Presto function
         int realParameterIndex = 0;
 
+        // Go through all the choices in the function and then pick the best one
+        List<ScalarImplementationChoice> choices = function.getAllChoices();
+        ScalarImplementationChoice bestChoice = choices.get(0);
+        for (int i = 0; i < choices.size(); i++) {
+            boolean isValid = false;
```

Comment:
The current implementation to decide the validity of implementation choice is incorrect. For existing scalar functions, it will consider the only implementation choice as invalid.

In fact, if you don't pre-assign `bestChoice` at the beginning, existing scalar functions will be unable to find a implementation choice.